docs/Explanation/ADR/adr-004-did.md

```diff
@@ -0,0 +1,208 @@
+# ADR 4: DID
+
+## Changelog
+
+- 2021-08-02: Initial draft
+
+## Status
+
+DRAFT
+
+
+## Abstract
+
+The ADR describes an implementation of the [W3C DID specification](https://www.w3.org/TR/did-core) 
+
+
+## Context
+
+The W3C DID specification are a building block to realize a SSI platform for tendermint based chains. 
+
+In the context of cosmos-cash project the SSI approach is a viable path to provide a technical solution for issues around identity, privacy and security that is compliant with regulations such as GDPR.
+
+## Decision
+
+> This section describes our response to these forces. It is stated in full sentences, with active voice. "We will ..."
+> {decision body}
+
+The DID W3C specification are designed using the "open world assumption" approach for data modelling. 
+For the cosmos-cash implementation we will be following the did-core specification: custom properties will require a fork and a different implementation of the cosmos-cash did module.
+
+### Method schema
+
+The schema for the cosmos-cash DID implementation is the following
+
+`did`:`cash`:`{CHAIN_NAME}`:`{ACCOUNT_ADDRESS (without chain prefix)}`
+
+i.e. `did:cash:cosmos:ts9ejqg7k4ht2sm53hycty875362yqxqmt9grj`
+
+where the `CHAIN_NAME` is used by a resolver as routing and the `ACCOUNT_ADDRESS` as the unique identifier for the DID
+
+> TODO: this is a placeholder, a decision must be taken in this regards, the method name could be more generically indicate the module like i.e. `tendermint`
+
+
+### DID Document
+
+The DID document will support the following attributes:
+
+- `context` - for json-ld specification
+- `id` - indicating the DID of the document
+- `controller` - a list of DID controllers for the DID document
+- `verificationMethods` - a list of verification method objects
+- `services` - a list of services
```

Comment:
**suggestion:** an example of a DID may be useful
**suggestion:** links to the terminology in the W3C specs would be beneficial as well.